x/ibc/23-commitment/types/merkle.go

```diff
@@ -104,6 +106,9 @@ func (p Path) String() string {
 // CONTRACT: provided path string MUST be a well formated path. See ICS24 for
 // reference.
 func ApplyPrefix(prefix exported.PrefixI, path string) Path {
+	if !validation.DefaultPathValidator(path) {
```

Comment:
Where should this check exist? Should it exist further up the call stack?

Reply:
It's ok here. We can prob return a `Path, error` and wrap the error from the validation